tests/integration/pilot/common/apps.go

```diff
@@ -255,6 +268,9 @@ func SetupApps(t resource.Context, i istio.Instance, apps *EchoDeployments) erro
 	if !t.Settings().SkipVM {
 		apps.VM = echos.Match(echo.Service(VMSvc))
 	}
+	if !t.Settings().SkipDelta {
+		apps.DeltaXDS = echos.Match(echo.Service(DeltaSvc))
```

Comment:
1. We don't need to deploy the app if this flag is on
2. "workload agnostic" tests won't filter this out

The only reason we don't deal with this for VMs is here:
https://github.com/istio/istio/blob/943ba0765876590d5c6da89d5df034fe0ea0808a/pkg/test/framework/components/echo/echoboot/echoboot.go#L98

Reply:
I fixed (1), I am not sure I understand how (2) works. if there are zero instances, then how/why are we sending traffic to it?